trie/trie.go

```diff
@@ -284,6 +285,12 @@ func (t *Trie) tryUpdate(key, value []byte) error {
 	return nil
 }
 
+// Arbitrum TODO: This was removed upstream, we should fix our code and remove it.
```

Comment:
agree with this TODO :)
Notice that DiskDB was removed from TrieDb but added to state.Database

The simplest solution seems to be adding a KeyValueStore recordingKV, and initializing it with something like:
```
  newRecordingKV(r.db.TrieDb(), r.db.DiskDB())
```

Reply:
Done.